File: RecallThat/Models/MemorySourceType.swift
import Foundation

/// Where a Memory item came from.
/// Only .screenshot is used in MVP. Other cases are reserved for Phase 10+ (Share Extension).
enum MemorySourceType: String, Codable, CaseIterable {
    case screenshot
    case sharedText    = "sharedText"
    case sharedURL     = "sharedURL"
    case sharedImage   = "sharedImage"
    case sharedPDF     = "sharedPDF"
    case sharedVideo   = "sharedVideo"
    case sharedAudio   = "sharedAudio"
    case sharedFile    = "sharedFile"   // generic doc, spreadsheet, presentation
}

File: RecallThatAndroid/app/src/main/java/com/recallthat/Models/MemorySourceType.kt
package com.recallthat.Models

enum class MemorySourceType(val raw: String) {
    SCREENSHOT("screenshot"),
    SHARED_TEXT("sharedText"),
    SHARED_URL("sharedURL"),
    SHARED_IMAGE("sharedImage"),
    SHARED_PDF("sharedPDF"),
    SHARED_VIDEO("sharedVideo"),
    SHARED_AUDIO("sharedAudio");

    companion object {
        fun from(raw: String): MemorySourceType =
            entries.firstOrNull { it.raw == raw } ?: SCREENSHOT
    }
}
